### PR TITLE
DOCSP-17218 what's new in 4.3

### PR DESCRIPTION
**Pull request by kyuan-mongodb**
## Pull Request Info
Added information on What's New in 4.3.

Reference used: https://mongodb.github.io/mongo-java-driver/4.3/whats-new/ 

Also added bullets for the following java tickets:
- https://jira.mongodb.org/browse/JAVA-3927
- https://jira.mongodb.org/browse/JAVA-3866
- https://jira.mongodb.org/browse/JAVA-4203

### Issue JIRA link:
https://jira.mongodb.org/browse/DOCSP-17218

### Docs staging link (requires sign-in on MongoDB Corp SSO):
https://docs-mongodbcom-staging.corp.mongodb.com/java/docsworker-xlarge/DOCSP-17218-WhatsNew4.3/whats-new/

### Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Does it render on staging correctly?
- [x] Are all the links working?
- [x] Are the staging links in the PR description updated?

### If your page documents a concept, does it meet the following criteria?

- [ ] Target the [Jasmin persona](https://drive.google.com/file/d/14FbBOLCVxwSP6M9BK4Nz1Ir9tzxT8_02/view)
- [ ] Target the [Lucas persona](https://drive.google.com/file/d/1J2vqJxo7ldv7OP_obA9Q-avf0o_ju4Lk/view)

### CODE REVIEW

**Review by terakilobyte — CHANGES_REQUESTED** [review]
This will need to incorporate the recent changes [here](https://github.com/mongodb/mongo-java-driver/pull/747/files#diff-bd319b7724ba91e97cc65d8d4d8d3bec052c6190388f122c7e41b36856b206d7R31-R33) and [here](https://github.com/mongodb/mongo-java-driver/blob/master/docs/reference/content/whats-new.md).

This ticket may be in a state of "waiting" up until .live

**Review by terakilobyte — APPROVED** [review]
lgtm

**Comment by kyuan-mongodb** [comment]
Hi @jyemin and @rozza, I want to make sure the content in this PR is technically correct.

**Review by rozza — APPROVED** [review]

File: source/whats-new.txt [hunk 1]
```diff
@@ -12,15 +12,64 @@ What's New
...
+What's New in 4.3
+-----------------
+
+New features of the 4.3 Java driver release include:
+
+- MongoDB Versioned API. For more information, see our :doc:`Versioned API guide </fundamentals/versioned-api>`. 
+- Added a ``setWindowFields`` pipeline stage to allow the use of window operators 
+- Added support for setting Netty `io.netty.handler.ssl.SslContext <https://netty.io/4.1/api/io/netty/handler/ssl/SslContext.html>`__
+- Added ``snapshot`` (non-speculative) as a ``readConcern`` level for read commands outside of transactions
+- The ``maxConnecting`` property limiting the number of connections you create in a connection pool
```
Comment (stIncMale) [inline source/whats-new.txt:31]: This a constant that cannot be changed neither via public nor via non-advertised means. I have doubts this change is supposed to be documented. But even if it is, we should only say that we limited the rate of establishing new TCP connections per server socket.

**Review by stIncMale — CHANGES_REQUESTED** [review]
I have only one concern. Please note that I do not have knowledge to review the changes with regard to removing restrictions on allowed characters.

**Review by stIncMale — APPROVED** [review]

File: source/whats-new.txt [hunk 2]
```diff
@@ -28,7 +28,7 @@ New features of the 4.3 Java driver release include:
 - Added a ``setWindowFields`` pipeline stage to allow the use of window operators 
 - Added support for setting Netty `io.netty.handler.ssl.SslContext <https://netty.io/4.1/api/io/netty/handler/ssl/SslContext.html>`__
 - Added ``snapshot`` (non-speculative) as a ``readConcern`` level for read commands outside of transactions
```
Comment (jyemin) [inline source/whats-new.txt:30]: This isn't completely accurate: that read concern level was actually added a long time ago.  The new feature is (indirect) support for `atClusterTime` for the read concern, which is supported via a new ClientSessionOption.

**Review by jyemin — CHANGES_REQUESTED** [review]

**Comment by jyemin** [comment]
@kyuan-mongodb I was in the process of reviewing this when you merged it.

**Pull request merged**